component/metrics/mutate/mutate.go

```diff
@@ -101,19 +101,22 @@ func (c *Component) Update(args component.Arguments) error {
 // series. This is a blocker for releasing a production-grade  of the metrics.mutate
 // component.
 func (c *Component) Receive(ts int64, metricArr []*metrics.FlowMetric) {
```

Comment:
Note for later:

It feels suspicious that this is an operation that never fails.

Reply:
It was hard to handle the failure, ie if it branches out N times and X succeed and Y fail what do you do? The decision was built around making it fire and forget, then have the individual components handle their own failures and not push those failures up the tree.

Reply:
Prometheus scrapers have logic that mark the scrape as failed if it can't write to the WAL. We're losing that capability here.

Reply:
Created issue #2035 for further discussion.